Sources/TuistKit/Services/Lint/LintCodeService.swift

```diff
@@ -0,0 +1,135 @@
+import Foundation
+import RxBlocking
+import TSCBasic
+import TuistCore
+import TuistGenerator
+import TuistLoader
+import TuistSupport
+
+enum LintCodeServiceError: FatalError, Equatable {
+    /// Thrown when neither a workspace or a project is found in the given path.
+    case manifestNotFound(AbsolutePath)
+    /// Thrown when target with given name does not exist.
+    case targetNotFound(String)
+    /// Throws when no lintable files found for target with given name.
+    case lintableFilesForTargetNotFound(String)
+
+    /// Error type.
+    var type: ErrorType {
+        switch self {
+        case .manifestNotFound, .targetNotFound, .lintableFilesForTargetNotFound:
+            return .abort
+        }
+    }
+
+    /// Description
+    var description: String {
+        switch self {
+        case let .manifestNotFound(path):
+            return "Couldn't find Project.swift nor Workspace.swift at \(path.pathString)"
+        case let .targetNotFound(name):
+            return "Target with name '\(name)' not found in the project."
+        case let .lintableFilesForTargetNotFound(name):
+            return "No lintable files for target with name '\(name)'"
+        }
+    }
+}
+
+final class LintCodeService {
+    private let rootDirectoryLocator: RootDirectoryLocating
+    private let codeLinter: CodeLinting
+    private let graphLoader: GraphLoading
+    private let manifestLoading: ManifestLoading
+
+    init(rootDirectoryLocator: RootDirectoryLocating = RootDirectoryLocator(),
+         codeLinter: CodeLinting = CodeLinter(),
+         manifestLoading: ManifestLoading = ManifestLoader(),
+         graphLoader: GraphLoading = GraphLoader(modelLoader: GeneratorModelLoader(manifestLoader: ManifestLoader(),
+                                                                                   manifestLinter: AnyManifestLinter())))
+    {
+        self.rootDirectoryLocator = rootDirectoryLocator
+        self.codeLinter = codeLinter
+        self.manifestLoading = manifestLoading
+        self.graphLoader = graphLoader
+    }
+
+    func run(path: String?, targetName: String?) throws {
+        // Determine destination path
+        let path = self.path(path)
+
+        // Load graph
+        let graph = try loadDependencyGraph(at: path)
+
+        // Get sources
+        let sources = try getSources(targetName: targetName, graph: graph)
+
+        // Lint code
+        logger.notice("Running code linting")
+        try codeLinter.lint(sources: sources, path: path)
+    }
+}
+
+// MARK: - Destination path
+
+private extension LintCodeService {
```

Comment:
Even though is a language feature, I'd avoid using extensions extensively because it difficults the work to the compiler. I'd rather keep everything as part of the same body definition and use `// MARK: -` annotations as you are already using.